plugins/gradle/tooling-extension-impl/src/org/jetbrains/plugins/gradle/tooling/util/resolve/DependencyResolverImpl.java

```diff
@@ -266,7 +266,7 @@ private FileCollection getCompileClasspath(SourceSet sourceSet) {
 
           ProjectComponentIdentifier projectComponentIdentifier = (ProjectComponentIdentifier)artifact.getId().getComponentIdentifier();
           String projectPath = projectComponentIdentifier.getProjectPath();
-          String key = projectPath + "_" + resolvedDependency.getConfiguration();
+          String key = projectComponentIdentifier.getProjectName() + "_" + resolvedDependency.getConfiguration();
```

Comment:
Proposed key will not be unique for projects with the same name but from different included builds.